The CPPFunction typedef (among others) have been deprecated in favour of
specific prototyped typedefs since readline 4.2.
It's been working since because compatibility typedefs have been
in place until they were removed in readline 6.3.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura parted-3.1.orig/parted/ui.c parted-3.1/parted/ui.c
--- parted-3.1.orig/parted/ui.c	2014-03-01 16:11:41.970827134 -0300
+++ parted-3.1/parted/ui.c	2014-03-01 16:11:55.540259786 -0300
@@ -1474,7 +1474,7 @@
 #ifdef HAVE_LIBREADLINE
   if (!opt_script_mode) {
     rl_initialize ();
-    rl_attempted_completion_function = (CPPFunction*) complete_function;
+    rl_attempted_completion_function = (rl_completion_func_t *) complete_function;
     readline_state.in_readline = 0;
   }
 #endif
